### DISCUSSION TITLE
x-transition in x-bind Alpine 3

### DISCUSSION THREAD

**Discussion by shiro1702** [alpinejs/alpine#1633] [7. Bugs (archived)]
hello!
if i try use this code in x-bind directive, alpine don't adding x-transition properties.

`
 
                    ['x-transition:enter'](){

                        return 'transition ease-in-out transform duration-300 absolute top-0 bottom-0 left-0 right-0'

                    },

                    ['x-transition:enter-start'](){

                        return  'opacity-0 -translate-x-full'

                    },

                    ['x-transition:enter-end'](){

                        return 'opacity-100 transform translate-x-0'

                    },

                    ['x-transition:leave'](){

                        return 'transition ease-in-out transform duration-300'

                    },

                    ['x-transition:leave-start'](){

                        return 'opacity-100 translate-x-0'

                    },

                    ['x-transition:leave-end'](){

                        return  'opacity-0 -translate-x-full'

                    }
`

**Comment by calebporzio** [comment]
Interesting, try just setting the value of the key to the string directly. For example:

```js
'x-transition:enter-end': 'opacity-100 transform translate-x-0',
},
```

**Reply by shiro1702** [reply]
thanks for your response.
but i need to set the value from a function dynamically.
like this

`
<section x-data="{ 
            tab: '1',
            oldTab: '1',
                way(val) {
                if (this.tab < this.oldTab) {
                    if ((''+val) == this.tab) {
                        return 'opacity-0 -translate-x-full'
                    }
                    if ((''+val) == this.oldTab) {
                        return 'opacity-0 translate-x-full'
                    }
                } else {
                    if ((''+val) == this.tab) {
                        return 'opacity-0 translate-x-full'
                    }
                    if ((''+val) == this.oldTab) {
                        return 'opacity-0 -translate-x-full'
                    }
                }
            },
            
            transitions(val) {
                return {
                    ['x-show']() {
                        return this.tab ===  '' + val
                    },
                    ['x-transition:enter'](){
                        return 'transition ease-in-out transform duration-300 absolute top-0 bottom-0 left-0 right-0'
                    },
                    ['x-transition:enter-start'](){
                        return this.way(val)
                    },
                    ['x-transition:enter-end'](){
                        return 'opacity-100 transform translate-x-0'
                    },
                    ['x-transition:leave'](){
                        return 'transition ease-in-out transform duration-300'
                    },
                    ['x-transition:leave-start'](){
                        return 'opacity-100 translate-x-0'
                    },
                    ['x-transition:leave-end'](){
                        return this.way(val)
                    }
                }
            },

        }"
        x-init="$nextTick(() => { $watch('tab', value => { console.log('value', value); $nextTick(() => { oldTab = value  }) }) }) "
>

    <div class="flex mx-auto border-2 border-indigo-500 rounded overflow-hidden mt-6">
        <button type="button" x-on:click="tab = '1'" :class="{ 'bg-indigo-500 text-white': tab === '1' }" class="py-1 px-4 focus:outline-none">индивидуальные</button>
        <button type="button" x-on:click="tab = '2'" :class="{ 'bg-indigo-500 text-white': tab === '2' }" class="py-1 px-4 focus:outline-none">мвлые группы</button>
        <button type="button" x-on:click="tab = '3'" :class="{ 'bg-indigo-500 text-white': tab === '3' }" class="py-1 px-4 focus:outline-none">группы</button>
    </div>
    


 <div class="relative">
   <div x-bind="transitions(1)" class="flex flex-wrap">
  </div>
   <div x-bind="transitions(2)" class="flex flex-wrap">
  </div>
   <div x-bind="transitions(3)" class="flex flex-wrap">
  </div>    
</div>
</section>
`

**Reply by calebporzio** [reply]
Here's an example of me dynamically setting x-transition values. It's actually from the Alpine homepage - hope this helps
![image](https://user-images.githubusercontent.com/3670578/123549191-e333e680-d735-11eb-8f32-bc6959310f3e.png)

**Reply by calebporzio** [reply]
I will say though - this is at the edge of intended behavior and maybe there's a more idiomatic solution like the one you're after probably.

**Reply by dennislizarzaburu** [reply]
Hi @calebporzio i was trying to achieve the same and found this issue. It looks like a good solution but doesn't work. Even your script from the Alpine homepage does't update the transition values. I always slides in from the right. Do you have any other idea how to dynamically update the classes on x-transition?

**Reply by julianwachholz** [reply]
Would love to see if this is possible too :)

**Reply by sritampatnaik** [reply]
Has anyone managed to find a solution to this?

**Reply by Tim-Wils** [reply]
#3051 is waiting for approval
